src/Core/LoginConfig/Builder/Handler/SwSsoLogin.php

```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\LoginConfig\Builder\Handler;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\LoginConfig\Builder\LoginConfigItem;
+use Symfony\Component\String\ByteString;
+
+/**
+ * @internal
+ */
+#[Package('core')]
+class SwSsoLogin extends AbstractLoginConfigHandler
+{
+    public function __construct(
+        private readonly string $appUrl,
+        private readonly string $admin,
+    ) {
+    }
+
+    public function createTemplateData(LoginConfigItem $loginConfigItem): array
+    {
+        return [
+            'key' => $loginConfigItem->key,
+            'snippet_key' => $loginConfigItem->snippetKey,
+            'icon' => $loginConfigItem->icon,
+            'class' => $loginConfigItem->class,
+            'url' => $this->createButtonUrl($loginConfigItem),
```

Comment:
If every "sso provider" has to return this info (because the frontend expects that, see your TS interface). I would suggest using a DTO for this and only leave `additionalData` as an mixed array (if that is even needed in the frontend)